src/chat/components/PreviewLink.tsx

```diff
@@ -0,0 +1,146 @@
+import React from 'react';
+import {
+  Linking,
+  StyleProp,
+  StyleSheet,
+  Text,
+  TouchableOpacity,
+  View,
+  ViewStyle,
+} from 'react-native';
+import { Bubble, BubbleProps } from 'react-native-gifted-chat';
+import type { MessageProps } from '../../interfaces';
+import { LinkPreview } from '@flyerhq/react-native-link-preview';
+
+export interface customPreviewLinkStyles {
+  customContainerStyle?: StyleProp<ViewStyle>;
+  customPreviewContainerStyle?: StyleProp<ViewStyle>;
+  customLinkTextStyle?: StyleProp<ViewStyle>;
+  customMessagePreviewStyle?: StyleProp<ViewStyle>;
+}
+interface PreviewLinkProps {
+  bubbleMessage: BubbleProps<MessageProps>;
+  customPreviewLinkStyles?: customPreviewLinkStyles;
+  customPreviewLink: (
+    urls: string[],
+    bubbleMessage: BubbleProps<MessageProps>
+  ) => JSX.Element;
+}
+
+export const PreviewLink: React.FC<PreviewLinkProps> = (props) => {
+  const { bubbleMessage, customPreviewLinkStyles, customPreviewLink } = props;
+  const { currentMessage } = bubbleMessage;
+  const urlRegex = /(https?:\/\/[^\s]+)/g;
+  const urls = currentMessage?.text.match(urlRegex);
+
+  const {
+    customContainerStyle,
+    customPreviewContainerStyle,
+    customLinkTextStyle,
+    customMessagePreviewStyle,
+  } = customPreviewLinkStyles || {};
+
+  const handleLinkPress = (url: string) => {
+    Linking.openURL(url).catch((err) =>
+      console.error('Error opening URL:', err)
+    );
+  };
+
+  const renderTextWithLinks = (text: string) => {
+    const parts = text.split(urlRegex);
+    return parts.map((part, index) => {
+      if (urlRegex.test(part)) {
+        return (
+          <TouchableOpacity key={index} onPress={() => handleLinkPress(part)}>
```

Comment:
Do we need `TouchableOpacity` for this?
Can we just add `onPress` into `Text` component below

Reply:
ok updated